File: gitdot-api-ts/src/endpoint/runner/verify-runner.ts
import { z } from "zod";

export const VerifyRunnerRequest = z.object({});
export type VerifyRunnerRequest = z.infer<typeof VerifyRunnerRequest>;

export const VerifyRunner = {
  path: "/ci/runner/{id}/verify",
  method: "POST",
  request: VerifyRunnerRequest,
  response: z.void(),
} as const;
export type VerifyRunner = typeof VerifyRunner;

File: gitdot-api/src/endpoint/runner/verify_runner.rs
use crate::endpoint::Endpoint;

pub struct VerifyRunner;

impl Endpoint for VerifyRunner {
    const PATH: &'static str = "/ci/runner/{id}/verify";
    const METHOD: http::Method = http::Method::POST;

    type Request = VerifyRunnerRequest;
    type Response = VerifyRunnerResponse;
}

pub type VerifyRunnerRequest = ();
pub type VerifyRunnerResponse = ();
